File: Data Structures & Algorithms/max-water-container/submission-2.cpp
class Solution {
public:
    int maxArea(vector<int>& nums) {
        int i = 0;
        int j = nums.size()-1;
        int res = 0;
        while(i < j) {
            int minAmongBoth = min(nums[i], nums[j]);
            res = max(res, minAmongBoth * (j - i));
            if(nums[i] >= nums[j])
                j--;
            else 
                i++;
        }
        return res;
    }
};

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-2.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        minToLeft = prices[0]
        maxProfit = 0
        for i in range(1, len(prices)):
            maxProfit = max(prices[i] - minToLeft, maxProfit)
            minToLeft = min(minToLeft, prices[i])
        return maxProfit
        
            

        